services/QuillLMS/app/controllers/teacher_infos_controller.rb

```diff
@@ -0,0 +1,36 @@
+class TeacherInfosController < ApplicationController
+
+  def create
+    teacher_info = TeacherInfo.create(
+      teacher: current_user,
+      minimum_grade_level: teacher_info_params[:minimum_grade_level],
+      maximum_grade_level: teacher_info_params[:maximum_grade_level]
+    )
+
+    subject_areas = SubjectArea.where(id: teacher_info_params[:subject_areas])
+    teacher_info.subject_areas.push(subject_areas)
+
+    render json: {}, status: 200
+  end
+
+  def update
+    teacher_info = TeacherInfo.find_or_create_by(teacher: current_user)
+
+    if teacher_info_params[:minimum_grade_level] && teacher_info_params[:maximum_grade_level]
+      teacher_info.update(minimum_grade_level: teacher_info_params[:minimum_grade_level], maximum_grade_level: teacher_info_params[:maximum_grade_level])
+    end
+
+    if teacher_info_params[:subject_areas]
+      teacher_info.teacher_info_subject_areas.destroy_all
+      subject_areas = SubjectArea.where(id: teacher_info_params[:subject_areas])
+      teacher_info.subject_areas.push(subject_areas)
```

Comment:
I don't really use `push` for object creation because of transparency, but instead opt for create! calls.  I would be curious to hear others thoughts on this.

Reply:
I did make that change because what you said makes sense to me, but I'm leaving the comment open for the sake of general discussion.

Reply:
Note: `push` doesn't necessarily create objects. If the pushed objects already exist, `push` has the effect of updating the foreign key value. 

Now, a Socratic dialogue:

Against: `ActiveRecord#push` does a lot more than `Array#push`, which can confound future debugging/diagnosing (what Brendan said).

For: This is a valid syntax which, while not used in our codebase, is prevalent in others. We should be permissive of introducing new coding styles. If we don't, our codebase can never stylistically improve. 

IMO `Against` wins, but it's debatable.

Reply:
Just for clarification, my point about transparency had to do with silent failure (i.e. there's no corresponding `push!`)  If a push fails for whatever reason, I don't think we would know about it.